### PR TITLE
rsx cache: Prevent crash on failure to obtain cache directory

### PR DESCRIPTION
**Pull request by elad335**
Similar to how PPU/SPU cache management does it.

### CODE REVIEW

File: rpcs3/Emu/RSX/rsx_cache.h [hunk 1]
```diff
@@ -197,16 +197,18 @@ namespace rsx
 			, pipeline_class_name(std::move(pipeline_class))
 			, m_storage(storage)
 		{
-			if (!g_cfg.video.disable_on_disk_shader_cache)
+			std::string cache_path = rpcs3::cache::get_ppu_cache();
```
Comment (Megamouse) [inline rpcs3/Emu/RSX/rsx_cache.h:200]:
```suggestion
			if (!g_cfg.video.disable_on_disk_shader_cache)
			{
				std::string cache_path = rpcs3::cache::get_ppu_cache();
				if (!cache_path.empty())
				{
					root_path = std::move(cache_path) + "shaders_cache/";
				}
			}
```

Comment (elad335) [reply rpcs3/Emu/RSX/rsx_cache.h]: Updated.

File: rpcs3/Emu/RSX/rsx_cache.h [hunk 2]
```diff
@@ -199,14 +199,17 @@ namespace rsx
 		{
 			if (!g_cfg.video.disable_on_disk_shader_cache)
 			{
-				root_path = rpcs3::cache::get_ppu_cache() + "shaders_cache";
+				if (std::string cache_path = rpcs3::cache::get_ppu_cache(); !cache_path.empty())
+				{
+					root_path = std::move(cache_path) + "shaders_cache/";
+				}
```
Comment (Megamouse) [inline rpcs3/Emu/RSX/rsx_cache.h:205]: It's probably better to throw an exception.
How is this even a use-case ?

Comment (elad335) [reply rpcs3/Emu/RSX/rsx_cache.h:205]: It is and I'm planning to use it.

Comment (Megamouse) [reply rpcs3/Emu/RSX/rsx_cache.h:205]: how

Comment (elad335) [reply rpcs3/Emu/RSX/rsx_cache.h:205]: Savestates debugging mode without requiring game files so no cache.

Comment (Megamouse) [reply rpcs3/Emu/RSX/rsx_cache.h:205]: then maybe add a fatal or an error here.

Comment (elad335) [reply rpcs3/Emu/RSX/rsx_cache.h:205]: It's logged in rpcs3::cache::get_ppu_cache() already.

Comment (Megamouse) [reply rpcs3/Emu/RSX/rsx_cache.h:205]: but only as a warning.
regular users won't see that

Comment (elad335) [reply rpcs3/Emu/RSX/rsx_cache.h:205]: Fixed.

**Pull request merged**